unified_planning/plans/contingent_plan.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unified_planning as up
+import unified_planning.plans as plans
+from typing import Callable, Dict, Optional, List, Tuple
+
+
+class ContingentPlanNode:
+    """This class represent a node in the tree contingent plan."""
+
+    def __init__(self, action_instance: "plans.plan.ActionInstance"):
+        self.action_instance = action_instance
+        self.children: List[
+            Tuple[Dict["up.model.FNode", "up.model.FNode"], "ContingentPlanNode"]
+        ] = []
+
+    def add_child(
```

Comment:
This method should check that the `env` of the added node is the same of the current node.